packages/grafana-schema/src/schema/dashboard/v2alpha0/dashboard.schema.cue

```diff
@@ -0,0 +1,500 @@
+  icon?: string
+  // Position in the mapping array. Only used internally.
+  index?: int32
+}
+
+// Color mode for a field. You can specify a single color, or select a continuous (gradient) color schemes, based on a value.
+// Continuous color interpolates a color using the percentage of a value relative to min and max.
+// Accepted values are:
+// `thresholds`: From thresholds. Informs Grafana to take the color from the matching threshold
+// `palette-classic`: Classic palette. Grafana will assign color by looking up a color in a palette by series index. Useful for Graphs and pie charts and other categorical data visualizations
+// `palette-classic-by-name`: Classic palette (by name). Grafana will assign color by looking up a color in a palette by series name. Useful for Graphs and pie charts and other categorical data visualizations
+// `continuous-GrYlRd`: ontinuous Green-Yellow-Red palette mode
+// `continuous-RdYlGr`: Continuous Red-Yellow-Green palette mode
+// `continuous-BlYlRd`: Continuous Blue-Yellow-Red palette mode
+// `continuous-YlRd`: Continuous Yellow-Red palette mode
+// `continuous-BlPu`: Continuous Blue-Purple palette mode
+// `continuous-YlBl`: Continuous Yellow-Blue palette mode
+// `continuous-blues`: Continuous Blue palette mode
+// `continuous-reds`: Continuous Red palette mode
+// `continuous-greens`: Continuous Green palette mode
+// `continuous-purples`: Continuous Purple palette mode
+// `shades`: Shades of a single color. Specify a single color, useful in an override rule.
+// `fixed`: Fixed color mode. Specify a single color, useful in an override rule.
+FieldColorModeId: "thresholds" | "palette-classic" | "palette-classic-by-name" | "continuous-GrYlRd" | "continuous-RdYlGr" | "continuous-BlYlRd" | "continuous-YlRd" | "continuous-BlPu" | "continuous-YlBl" | "continuous-blues" | "continuous-reds" | "continuous-greens" | "continuous-purples" | "fixed" | "shades" @cuetsy(kind="enum",memberNames="Thresholds|PaletteClassic|PaletteClassicByName|ContinuousGrYlRd|ContinuousRdYlGr|ContinuousBlYlRd|ContinuousYlRd|ContinuousBlPu|ContinuousYlBl|ContinuousBlues|ContinuousReds|ContinuousGreens|ContinuousPurples|Fixed|Shades")
+
+// Defines how to assign a series color from "by value" color schemes. For example for an aggregated data points like a timeseries, the color can be assigned by the min, max or last value.
+FieldColorSeriesByMode: "min" | "max" | "last"
+
+// Map a field to a color.
+FieldColor: {
+    // The main color scheme mode.
+    mode: FieldColorModeId
+    // The fixed color value for fixed or shades color modes.
+    fixedColor?: string
+    // Some visualizations need to know how to assign a series color from by value color schemes.
+    seriesBy?: FieldColorSeriesByMode
+}
+
+// Dashboard Link type. Accepted values are dashboards (to refer to another dashboard) and link (to refer to an external resource)
+DashboardLinkType: "link" | "dashboards"
```

Comment:
I couldn't find how to do this in CUE and Cog. Or even if we need it.

In the end, the enums can cover that scenario, so they should be equivalent and we can easily migrate to them

Reply:
thank you, this is good to know! 😄

Reply:
The fact that `DashboardLinkType` are now generated as an enum is coming directly from cog: anything like `"link" | "dashboards"` will be converted to an enum instead of a "type" (this was done mostly because the alternative is keeping the union type and that's problematic in other languages than Typescript)

If this is causing issues, we can probably find a way to annotate the schema to say "hey, generate a TS type here, not an enum"

Reply:
+1 for using enums whenever possible